### PR TITLE
Switch from JSHint to ESLint

### PR DESCRIPTION
**Pull request by mattmahn**
### Summary

We're switching to ESLint because TheFourFifths/consus-client will be using React/JSX, and JSHint does not support that. To make things simpler, we'll use the same linter (ESLint) across all projects.

### Checklist

- [x] Did you run `npm test`?
- [x] Did you update/add documentation for new methods or changed functionality?

Finally, is it ready to be reviewed and merged: YES :white_check_mark: 

### How to Review

1. Verify that ESLint is being used by looking in `package.json`/`Gruntfile.js`
1. Run `grunt lint`

### Links

- [TFF-51](https://msoese.atlassian.net/browse/TFF-51)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "rules": {
+        "indent": [
+            2,
+            4
+        ],
+        "quotes": [
+            2,
+            "single"
+        ],
+        "linebreak-style": [
```
Comment (thofmann) [inline .eslintrc.json]: Let's remove linebreak styles on this one, too.

https://github.com/TheFourFifths/consus-flux/pull/9/commits/06e9f2992ede08b94495201fd3b35f72256d7e7d

File: src/index.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ let window;
 // Create the main app window
 function createWindow() {
     // Create the browser window
-    let window = new BrowserWindow({
```
Comment (thofmann) [inline src/index.js]: :+1:

File: src/store/authentication-store.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import { Store } from 'consus-flux';
 
+/* eslint-disable no-unused-vars */
```
Comment (thofmann) [inline src/store/authentication-store.js]: This is fine for now since this is basically a stubbed class, but let's try to avoid disabling the linter.

Comment (mattmahn) [inline src/store/authentication-store.js]: Of course, that was my thinking exactly.

File: .eslintrc.json (hunk 1 above)
Comment (mattmahn) [inline .eslintrc.json]: removed

**Pull request merged**